[DOC] Improve documentation about podAntiAffinity for Kafka brokers (#5017)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/proc-scheduling-brokers-on-different-worker-nodes.adoc

// Module included in the following assemblies:
//
// assembly-scheduling.adoc

[id='configuring-pod-anti-affinity-to-schedule-each-kafka-broker-on-a-different-worker-node-{context}']
= Configuring pod anti-affinity to schedule each Kafka broker on a different worker node

> When multiple Kafka brokers or multiple ZooKeeper nodes are running on the same Kubernetes worker nodes, they will all become unavailable at the same time in case of worker node issues.

Reviewer: ```suggestion
Many Kafka brokers or  ZooKeeper nodes can run on the same Kubernetes worker node. If the worker node fails, they will all become unavailable at the same time. 
```

> To increase the reliability, you can use `podAntiAffinity` to schedule each Kafka broker or ZooKeeper node on different Kubernetes worker node.

Reviewer: ```suggestion
To improve reliability, you can use `podAntiAffinity` configuration to schedule each Kafka broker or ZooKeeper node on a different Kubernetes worker node.
```

.Prerequisites

* A Kubernetes cluster
* A running Cluster Operator

.Procedure

. Edit the `affinity` property in the resource specifying the cluster deployment.

> To make sure that no worker nodes are shared by Kafka brokers or by ZooKeeper nodes, use the `strimzi.io/name` label.

Reviewer: ```suggestion
To make sure that no worker nodes are shared by Kafka brokers or ZooKeeper nodes, use the `strimzi.io/name` label.
```

> The `topologyKey` should be set to `kubernetes.io/hostname` to specify that the selected pods should not be scheduled on nodes with the same hostname.

Reviewer: ```suggestion
Set the `topologyKey` to `kubernetes.io/hostname` to specify that the selected pods are not scheduled on nodes with the same hostname.
```

For example:
+
[source,yaml,subs="+quotes,attributes+"]
----
apiVersion: {KafkaApiVersion}
kind: Kafka
spec:
  kafka:
    # ...
    template:
      pod:
        affinity:
          podAntiAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              - labelSelector:
                  matchExpressions:
                    - key: strimzi.io/cluster
                      operator: In
                      values:
                        - _CLUSTER-NAME_
                topologyKey: "kubernetes.io/hostname"
    # ...
  zookeeper:
    # ...
    template:
      pod:
        affinity:
          podAntiAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              - labelSelector:
                  matchExpressions:
                    - key: strimzi.io/cluster
                      operator: In
                      values:
                        - _CLUSTER-NAME_
                topologyKey: "kubernetes.io/hostname"
    # ...
----
+
Where `_CLUSTER-NAME_` is the name of your Kafka custom resource.

> . Create or update the resource.
> +
> This can be done using `kubectl apply`:
> [source,shell,subs=+quotes]
> kubectl apply -f _KAFKA-CONFIG-FILE_

Reviewer: ```suggestion
. Create or update the resource.
+
[source,shell,subs=+quotes]
kubectl apply -f _KAFKA-CONFIG-FILE_
```

Reviewer: We stopped including the _This can be done..._
